tests/Serializer/GraphNavigatorTest.php

```diff
@@ -131,6 +132,37 @@ public function testNavigatorChangeTypeOnSerialization()
         $navigator->accept($object, null);
     }
 
+    public function testExposeAcceptHandlerExceptionOnSerialization()
+    {
+        $object = new SerializableClass();
+        $typeName = 'JsonSerializable';
+        $format = 'foo';
+        $msg = 'Useful serialization error with relevant context information';
+
+        $handler = static function ($visitor, $data, array $type, SerializationContext $context) use ($msg) {
+            $context->startVisiting(new \stdClass());
+            throw new \RuntimeException($msg);
+        };
+        $this->handlerRegistry->registerHandler(GraphNavigatorInterface::DIRECTION_SERIALIZATION, $typeName, $format, $handler);
+
+        $visitor = $this->createMock(VisitorInterface::class);
+        $context = $this->getMockBuilder(SerializationContext::class)
+            ->enableOriginalConstructor()
+            ->setMethodsExcept(['initialize', 'startVisiting', 'stopVisiting'])
+            ->getMock();
+        $context->method('getFormat')->willReturn($format);
+
+        $navigator = new SerializationGraphNavigator($this->metadataFactory, $this->handlerRegistry, $this->accessor, $this->dispatcher);
+        $navigator->initialize($this->serializationVisitor, $context);
+        $context->initialize($format, $visitor, $navigator, $this->metadataFactory);
+
+        try {
+            $navigator->accept($object, ['name' => $typeName, 'params' => []]);
+        } catch (\RuntimeException $exception) {
+            $this->assertEquals($msg, $exception->getMessage());
```

Comment:
can you use `self::expectedException()` and self::expectedExceptionMessage()`  from phpunit instead of the current approach. This test will be green if for some reason the exception is not thrown.

Reply:
Good point!